File: 206. Reverse Linked List/iterative_solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
    curr := head
    var tail *ListNode
    for curr != nil {
        tmp := curr
        curr = curr.Next
        tmp.Next = tail
        tail = tmp
    }
    return tail
}

File: 206. Reverse Linked List/iterative_solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        tail = None
        while head:
            tmp = head.next
            head.next = tail
            tail = head
            head = tmp
            
        return tail
